### PR TITLE
Updated to explain the library work

### PR DESCRIPTION
**Pull request by smaddox-sf**
Reworked this page to explain overall process with libraries and explain the VS Code Settings for Apex & Deploy / Retrieve

@W-9027798@

### CODE REVIEW

**Comment by smaddox-sf** [comment]
@ekapner - PR for the updates to the "performance enhancements" page

File: docs/_articles/en/user-guide/perf-enhancements.md [hunk 1]
```diff
@@ -3,21 +3,34 @@ title: Performance Enhancements
 lang: en
 ---
 
-The changes we have made to enhance performance ensure that metadata deploys and retrieves run through a more efficient code path.
+The Apex, Deploy and Retrieve commands are in the process of moving to a new architecture.  These commands have been rearchitected into independent Typescript libraries and CLI plugins.  This is part of our larger effort to break up the salesforce-alm plug-in.  [You can read more about that in this blog post](https://developer.salesforce.com/blogs/2021/02/open-sourcing-salesforce-cli-update-feb-2021.html).  There’s many great benefits to this effort and one of the best is better performance.  Especially for single file deploys & retrieves, you should notice a significant performance gain when using the latest architecture.
```
Comment (ekapner) [inline docs/_articles/en/user-guide/perf-enhancements.md:6]:
```suggestion
The Apex, Deploy and Retrieve commands are in the process of moving to a new architecture. These commands have been rearchitected into independent Typescript libraries and CLI plugins as part of our larger effort to break up the salesforce-alm plugin.  [You can read more about that in this blog post](https://developer.salesforce.com/blogs/2021/02/open-sourcing-salesforce-cli-update-feb-2021.html). There’s many great benefits to this effort and one of the best is better performance. For single file deploys and retrieves, you'll likely notice a significant performance gain when using the latest architecture.
```

File: docs/_articles/en/user-guide/perf-enhancements.md [hunk 2]
```diff
@@ -3,21 +3,34 @@ title: Performance Enhancements
...
 
-> NOTICE: The changes to enhance code performance are currently in beta. If you find any bugs or have feedback, [open a GitHub issue](./en/bugs-and-feedback).
+## Execution Process & Status
```
Comment (ekapner) [inline docs/_articles/en/user-guide/perf-enhancements.md:8]:
```suggestion
## Execution Process and Status
```

File: docs/_articles/en/user-guide/perf-enhancements.md [hunk 3]
```diff
@@ -3,21 +3,34 @@ title: Performance Enhancements
...
+
+This architecture change is done through a phased approach.  For both the Apex library and Deploy/Retrieve library, we are rolling out the changes through these phases:
+
+1. Library created & completed
```
Comment (ekapner) [inline docs/_articles/en/user-guide/perf-enhancements.md:12]:
```suggestion
1. Library created and completed.
```

File: docs/_articles/en/user-guide/perf-enhancements.md [hunk 4]
```diff
@@ -3,21 +3,34 @@ title: Performance Enhancements
...
+2. Available to try it in VS Code - We provide a VS Code setting so you can opt in and try it out. 
+3. On by default in VS Code - The VS code setting default is defaulted to True to enable the library.  If you previously opted in and then opted out, you will have to manually opt back in to use the feature.  We change the default after the commands have been available and used for months.
```
Comment (ekapner) [inline docs/_articles/en/user-guide/perf-enhancements.md:14]:
```suggestion
3. On by default in VS Code - The VS code setting default is defaulted to True to enable the library. If you previously opted in and then opted out, you'll have to manually opt back in to use the feature.  We'll change the default after the commands have been available and used for months.
```

File: docs/_articles/en/user-guide/perf-enhancements.md [hunk 5]
```diff
@@ -3,21 +3,34 @@ title: Performance Enhancements
...
+|                         | Library complete	| Available to try in VS Code	| On by default in VS Code	| CLI Updated  |
+|-------------------------|:-----------------:|:---------------------------:|:-------------------------:|:------------:|
+| Apex Library	          |        ✔️          |               ✔️             |             ✔️             |              |
+| Deploy Retrieve Library |        ✔️          |               ✔️             |                           |              |
+	
 
 ## Setup
 
-To enable this beta feature:
+There is a VS Code setting for each library.  Access them by selecting Select *File* > *Preferences* > *Settings* (Windows or Linux) or *Code* > *Preferences* > *Settings* (macOS)
```
Comment (ekapner) [inline docs/_articles/en/user-guide/perf-enhancements.md:29]:
```suggestion
There is a VS Code setting for each library.  Access them by selecting Select **File** > **Preferences** > **Settings** (Windows or Linux) or **Code** > **Preferences** > **Settings** (macOS).
```

Comment (ekapner) [reply docs/_articles/en/user-guide/perf-enhancements.md:29]: I think we bold menu selections.

**Review by ekapner — COMMENTED** [review]
A few edits

File: docs/_articles/en/user-guide/perf-enhancements.md [hunk 6]
```diff
@@ -3,21 +3,34 @@ title: Performance Enhancements
...
-Performance enhancements are in effect when you run the following commands:
+1. *Experimental: Deploy Retrieve*.
+2. *Experimental: Use Apex Library*
 
-- SFDX: Deploy This Source to Org
-- SFDX: Retrieve This Source from Org
-- SFDX: Deploy Source in Manifest to Org
-- SFDX: Retrieve Source in Manifest from Org
-- Retrieve Using the [Org Browser](./en/user-guide/org-browser)
+This work is in process, so please [open a GitHub issue](https://github.com/forcedotcom/salesforcedx-vscode/issues/new/choose) if you find any bugs or have feedback.
```
Comment (smaddox-sf) [inline docs/_articles/en/user-guide/perf-enhancements.md:36]:
```suggestion
Known gaps:
- If you're using [CLI Plug-In Hooks](https://developer.salesforce.com/docs/atlas.en-us.sfdx_cli_plugins.meta/sfdx_cli_plugins/cli_plugins_customize_hooks.htm), those are not available when using the library from VS Code.  Uncheck the settings above to use the CLI directly for those commands.

If you spot other gaps or have feedback, please [open a GitHub issue](https://github.com/forcedotcom/salesforcedx-vscode/issues/new/choose) if you find any bugs or have feedback.
```

File: docs/_articles/en/user-guide/perf-enhancements.md [hunk 7]
```diff
@@ -3,21 +3,37 @@ title: Performance Enhancements
...
-Performance enhancements are in effect when you run the following commands:
+Known gaps:
+- If you're using [CLI Plug-In Hooks](https://developer.salesforce.com/docs/atlas.en-us.sfdx_cli_plugins.meta/sfdx_cli_plugins/cli_plugins_customize_hooks.htm), those are not available when using the library from VS Code.  Uncheck the settings above to use the CLI directly for those commands.
 
-- SFDX: Deploy This Source to Org
-- SFDX: Retrieve This Source from Org
-- SFDX: Deploy Source in Manifest to Org
-- SFDX: Retrieve Source in Manifest from Org
-- Retrieve Using the [Org Browser](./en/user-guide/org-browser)
+If you spot other gaps or have feedback, please [open a GitHub issue](https://github.com/forcedotcom/salesforcedx-vscode/issues/new/choose) if you find any bugs or have feedback.
```
Comment (smaddox-sf) [inline docs/_articles/en/user-guide/perf-enhancements.md:39]:
```suggestion
If you spot other gaps or have feedback, please [open a GitHub issue](https://github.com/forcedotcom/salesforcedx-vscode/issues/new/choose).
```

**Pull request merged**